### PR TITLE
New Resource : `azurerm_storage_mover_project`

### PR DESCRIPTION
**Pull request by sinbai**
Swagger: https://github.com/Azure/azure-rest-api-specs/blob/add28efcd3a5fd422285d992fb1ec5ee5a7a40a6/specification/storagemover/resource-manager/Microsoft.StorageMover/stable/2023-03-01/storagemover.json#L916

Test Results:
image

### CODE REVIEW

File: internal/services/storagemover/storage_mover_project_resource.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			properties := resp.Model
+			if properties == nil {
+				return fmt.Errorf("retrieving %s: model was nil", id)
+			}
+
+			if metadata.ResourceData.HasChange("description") {
+				properties.Properties.Description = &model.Description
```
Comment (mbfrahry) [inline internal/services/storagemover/storage_mover_project_resource.go:140]: `properties.Properties` could be nil so we need to check that it isn't before trying to set `properties.Properties.Description`.

`properties.Properties.Description = &model.Description`

to

`properties.Properties.Description = pointer.To(model.Description)`

File: internal/services/storagemover/storage_mover_project_resource.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+			}
+
+			state := StorageMoverProjectResourceModel{
+				Name:           id.ProjectName,
+				StorageMoverId: storagemovers.NewStorageMoverID(id.SubscriptionId, id.ResourceGroupName, id.StorageMoverName).ID(),
+			}
+
+			if resp.Model != nil {
+				if properties := resp.Model.Properties; properties != nil {
+					if properties.Description != nil {
```
Comment (mbfrahry) [inline internal/services/storagemover/storage_mover_project_resource.go:179]: This if block can be replaced with 
```suggestion
					state.Description = pointer.From(properties.Description)
```

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @sinbai, this PR looks good! Just need another nil check and a simplified way to set a variable. Once those and the merge conflicts are fixed, we'll be good to go!

File: internal/services/storagemover/storage_mover_project_resource.go (hunk 2 above)
Comment (sinbai) [reply internal/services/storagemover/storage_mover_project_resource.go]: Fixed.

File: internal/services/storagemover/storage_mover_project_resource.go (hunk 1 above)
Comment (sinbai) [reply internal/services/storagemover/storage_mover_project_resource.go]: Fixed.

**Comment by sinbai** [comment]
> Hey @sinbai, this PR looks good! Just need another nil check and a simplified way to set a variable. Once those and the merge conflicts are fixed, we'll be good to go!

Hi @mbfrahry, thanks for  your feedback. I have updated the code. Could you please take another look?

Test results:
image

**Review by mbfrahry — APPROVED** [review]
LGTM! Thanks for those changes @sinbai

**Pull request merged**